src/java/org/jivesoftware/openfire/net/SocketConnection.java

```diff
@@ -145,11 +141,11 @@ public SocketConnection(PacketDeliverer backupDeliverer, Socket socket, boolean
         // DANIELE: Modify socket to use channel
         if (socket.getChannel() != null) {
             writer = Channels.newWriter(
-                    ServerTrafficCounter.wrapWritableChannel(socket.getChannel()), CHARSET);
+                    ServerTrafficCounter.wrapWritableChannel(socket.getChannel()), StandardCharsets.UTF_8.newEncoder(), -1);

```

Comment:
Yes, it is. There's no Channels.newWriter(Channel, Charset). The -1 is correct as well (taken from Channels.newWriter(Channel, String).
Thanks for reviewing!